Failing for java 8 update 252

**Describe the bug**
We have java version `Java version: 1.8.0_252` but it still fails saying `dependency-check requires Java 8 update 251 or higher`

**Version of dependency-check used**
The problem occurs using version 9.0.1 of the maven plugin.

**Log file**

```bash
go@myserver:/var/lib/go-agent-2/pipelines/demo-hello-world$ mvn --version
Apache Maven 3.6.0
Maven home: /usr/share/maven
Java version: 1.8.0_252, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-8-openjdk-amd64/jre
Default locale: en_US, platform encoding: ANSI_X3.4-1968
OS name: "linux", version: "4.19.0-25-cloud-amd64", arch: "amd64", family: "unix"
go@integration-agarg:/var/lib/go-agent-2/pipelines/demo-hello-world$ mvn --batch-mode clean org.owasp:dependency-check-maven:9.0.1:check verify -Pcoverage -DossindexAnalyzerEnabled=false -DnvdApiKey=xxxx
[INFO] Scanning for projects...
[INFO]
[INFO] --------------------< com.oscato:demo-hello-world >---------------------
[INFO] Building demo-hello-world 0.7.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-clean-plugin:3.1.0:clean (default-clean) @ demo-hello-world ---
[INFO]
[INFO] --- dependency-check-maven:9.0.1:check (default-cli) @ demo-hello-world ---
[ERROR] Non-supported Java Runtime: dependency-check requires at least Java 8 update 251 or higher.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.572 s
[INFO] Finished at: 2023-11-28T09:58:11Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.owasp:dependency-check-maven:9.0.1:check (default-cli) on project demo-hello-world: Execution default-cli of goal org.owasp:dependency-check-maven:9.0.1:check failed: dependency-check requires Java 8 update 251 or higher -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
```

**To Reproduce**
Steps to reproduce the behavior:

Run: `mvn --batch-mode clean org.owasp:dependency-check-maven:9.0.1:check verify -Pcoverage -DossindexAnalyzerEnabled=false -DnvdApiKey=xxxx` command in any maven project.


**Expected behavior**
Should pass or give a different error as we already have a higher version than minimum required for java 8.

--- a/core/src/main/java/org/owasp/dependencycheck/utils/Utils.java
+++ b/core/src/main/java/org/owasp/dependencycheck/utils/Utils.java
@@ -17,11 +17,20 @@
  */
 package org.owasp.dependencycheck.utils;
 
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 /**
  *
  * @author Jeremy Long
  */
 public final class Utils {
+
+    /**
+     * The logger.
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger(Utils.class);
 
     /**
      * Empty constructor for utility class.
@@ -55,9 +64,20 @@ public final class Utils {
     public static int getJavaUpdateVersion() {
         //"1.8.0_144" "11.0.2+9" "17.0.8.1"
         String runtimeVersion = System.getProperty("java.runtime.version");
+        return parseUpdate(runtimeVersion);
+    }
+
+    /**
+     * Parses the update version from the runtime version.
+     *
+     * @param runtimeVersion the runtime version
+     * @return the update version
+     */
+    protected static int parseUpdate(String runtimeVersion) {
+        LOGGER.debug(runtimeVersion);
         try {
             String[] parts = runtimeVersion.split("\\.");
-            if (parts.length == 4) {
+            if (parts.length == 4 && isNumeric(parts)) {
                 return Integer.parseInt(parts[2]);
             }
             int pos = runtimeVersion.indexOf('_');
@@ -68,9 +88,9 @@ public final class Utils {
                     return 0;
                 }
             }
-            int end = runtimeVersion.lastIndexOf('+');
+            int end = runtimeVersion.indexOf('+', pos);
             if (end < 0) {
-                end = runtimeVersion.lastIndexOf('-');
+                end = runtimeVersion.indexOf('-', pos);
             }
             if (end > pos) {
                 return Integer.parseInt(runtimeVersion.substring(pos + 1, end));
@@ -82,8 +102,26 @@ public final class Utils {
         }
     }
 
+    /**
+     * Determines if all parts of the string array are numeric.
+     *
+     * @param parts the strings to check
+     * @return true if all of the strings in the array are numeric; otherwise
+     * false
+     */
+    private static boolean isNumeric(String[] parts) {
+        for (String i : parts) {
+            if (!StringUtils.isNumeric(i)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
     public static void main(String[] args) {
+        System.out.println("Java runtime : " + System.getProperty("java.runtime.version"));
         System.out.println("Java version : " + getJavaVersion());
         System.out.println("Java update  : " + getJavaUpdateVersion());
+
     }
 }

--- a/core/src/test/java/org/owasp/dependencycheck/utils/UtilsTest.java
+++ b/core/src/test/java/org/owasp/dependencycheck/utils/UtilsTest.java
@@ -1,0 +1,56 @@
+/*
+ * This file is part of dependency-check-core.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright (c) 2023 Jeremy Long. All Rights Reserved.
+ */
+package org.owasp.dependencycheck.utils;
+
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+/**
+ *
+ * @author Jeremy Long
+ */
+public class UtilsTest {
+
+    /**
+     * Test of parseUpdate method, of class Utils.
+     */
+    @Test
+    public void testParseUpdate() {
+
+        String runtimeVersion = "1.8.0_252-8u252-b09-1~deb9u1-b09";
+        int expResult = 252;
+        int result = Utils.parseUpdate(runtimeVersion);
+        assertEquals(expResult, result);
+
+        runtimeVersion = "1.8.0_144";
+        expResult = 144;
+        result = Utils.parseUpdate(runtimeVersion);
+        assertEquals(expResult, result);
+
+        runtimeVersion = "11.0.2+9";
+        expResult = 2;
+        result = Utils.parseUpdate(runtimeVersion);
+        assertEquals(expResult, result);
+
+        runtimeVersion = "17.0.8.1";
+        expResult = 8;
+        result = Utils.parseUpdate(runtimeVersion);
+        assertEquals(expResult, result);
+    }
+
+}

